src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/cachemanager/ComputableCacheNode.java

```diff
@@ -0,0 +1,165 @@
+package org.broadinstitute.hellbender.tools.coveragemodel.cachemanager;
+
+import org.broadinstitute.hellbender.utils.Utils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * This class represents an automatically computable cache node
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class ComputableCacheNode extends CacheNode {
+
+    private final boolean cacheEvals;
```

Comment:
OK, we discussed some of this in person.  Let's file an issue and try to see if we can refactor!